File: AppSets-Android/appsets/src/main/kotlin/xcj/app/appsets/util/message_digest/MD5Helper.kt
package xcj.app.appsets.util.message_digest

import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import xcj.app.starter.android.util.PurpleLogger
import xcj.app.starter.util.ByteUtil
import java.security.MessageDigest
import kotlin.io.encoding.Base64
import kotlin.io.encoding.ExperimentalEncodingApi

object MD5Helper {

    private const val TAG = "MD5Helper"

    data class TransformResult(
        val outContent: String,
        val outContentBase64: String
    )

    @JvmStatic
    @OptIn(ExperimentalEncodingApi::class)
    suspend fun transform(input: String): TransformResult? {
        return withContext(Dispatchers.Default) {
            runCatching {
                val messageDigest = MessageDigest.getInstance("MD5")
                val outBytes = messageDigest.digest(input.toByteArray())
                val outContent = ByteUtil.toHexString(outBytes)
                val outContentBase64 = Base64.encode(outBytes)
                return@withContext TransformResult(outContent, outContentBase64)
            }.onFailure {
                PurpleLogger.current.e(TAG, "encrypt failed!, ${it.message}")
            }
            PurpleLogger.current.d(TAG, "encrypt failed!")
            return@withContext null
        }
    }
}

File: AppSets-iOS/AppSets/com/app/appsets/util/messge_digest/MD5Helper.swift
//
//  MD5Helper.swift
//  AppSets
//
//  Created by caiju Xu on 2024/7/9.
//

import Foundation
import CryptoSwift

struct MD5Helper {
    
    struct TransformResult {
        let outContent:String
        let ountConentBase64:String
    }
    
    static func transform(_ input:String) -> TransformResult {
        let outContent = input.md5()
        let outContentBase64 = outContent.data(using: .utf8)!.base64EncodedString()
        return TransformResult(outContent: input.md5(), ountConentBase64: outContentBase64)
    }
    
}
